src/cares_wrap.cc

```diff
@@ -1797,22 +1797,24 @@ static void Query(const FunctionCallbackInfo<Value>& args) {
 
   Local<Object> req_wrap_obj = args[0].As<Object>();
   Local<String> string = args[1].As<String>();
-  Wrap* wrap = new Wrap(channel, req_wrap_obj);
+  std::unique_ptr<Wrap> wrap {new Wrap(channel, req_wrap_obj)};
```

Comment:
Could you use `std::make_unique` (since this uses `new` and not `Malloc`)?

Reply:
This should be managed with an RAII wrapper not `unique_ptr`.
```suggested
  Wrap* wrap = new Wrap(channel, req_wrap_obj);
  OnScopeLeave([wrap](){ delete wrap; })
```
... and later
```cpp
if (err == 0) {
  OnScopeLeave.cancel()
}
```

Reply:
The reason for not using [std::make_unique](https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique) is that I thought that we still have compilers that don't support c++14, and have to stick with c++11 features. If this is not correct I'm happy to use it. Thanks.

Reply:
I noticed that we can use std::make_unique. Thanks!